index/scorch/segment/zap/segment.go

```diff
@@ -418,38 +418,29 @@ func (s *SegmentBase) DocNumbers(ids []string) (*roaring.Bitmap, error) {
 		}
 
 		postingsList := emptyPostingsList
-
-		skipCheck := false
+		filteredIds := ids[:0]
 		sMax := ""
-		iMin := ""
 
-		if len(ids) > 0 {
-			sMaxB, err := idDict.fst.GetMaxKey()
-			if err != nil {
-				skipCheck = true
-			} else {
-				sMax = string(sMaxB)
-				iMin = ids[0]
-				for i := 1; i < len(ids); i++ {
-					if ids[i] < iMin {
-						iMin = ids[i]
-					}
+		sMaxB, err := idDict.fst.GetMaxKey()
```

Comment:
would it be simpler to return the err right here if err != nil?

if the fst is empty, what does GetMaxKey() return?  "", nil?

If so, then sMaxB ought to be lower than any real id, so that might be an edge case that "falls out for free"?

Reply:
Yes, now fixed.